google_compute_engine_oslogin/nss_module/nss_oslogin.cc

```
@@ -57,7 +57,9 @@ int _nss_oslogin_getpwuid_r(uid_t uid, struct passwd *result, char *buffer,
   std::stringstream url;
   url << kMetadataServerUrl << "users?uid=" << uid;
   string response;
-  if (!HttpGet(url.str(), &response) || response.empty()) {
+  long http_code = 0;
+  if (!HttpGet(url.str(), &response, &http_code) || http_code >= 400 ||
```

We should probably also return an error for anything other than 200 here.